packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -735,9 +748,32 @@ class Navigator extends StatefulWidget {
   /// }
   /// ```
   /// {@end-tool}
+  ///
+  /// {@tool sample}
+  ///
+  /// The following example shows how to pass additional `arguments` to the
+  /// route:
+  ///
+  /// ```dart
+  /// void _didPushButton() {
+  ///   Navigator.pushNamed(
+  ///     context,
+  ///     '/weather',
+  ///     arguments: <String, String>{
+  ///       'city': 'Berlin',
+  ///       'country': 'Germany',
+  ///     },
+  ///   );
+  /// }
+  /// ```
+  /// {@end-tool}
```

Comment:
I think it would be worth having a sample that passes an object with methods on it, pointing out that there's no serialization going on and that the methods will be preserved.  Maybe something like


````
```dart
class WeatherRouteArguments {
  WeatherRouteArguments(this.city, this.country);
  final String city;
  final String country;

  bool isCapitolCity() {
    if (country == 'Germany') {
      return city == 'Berlin';
    }
  }
}

void _didPushButton() {
  Navigator.pushNamed(
    context,
    '/weather',
    arguments: WeatherRouteArguments(city: 'Berlin', country: 'Germany'),
  );
}
```
````

Or, if nothing else, to give a hint.  I have a feeling that people would normally be better off using classes than maps.